### ISSUE TITLE
Error downloading a key with Avahi

### ISSUE THREAD

**Issue by RyuzakiKK** [gnome-keysign/gnome-keysign#90]
Steps to reproduce:
- Leave the internet button disabled
- Choose a key to send
- Open another instance in the receive tab
- Copy and paste the key fingerprint to receive the key

Instead of receiving the key this error happens:
```
keysign.util (DEBUG): Parsed barcode into {'fingerprint': '1354 596C D8C7 90AD D9B2\n7D70 90C7 9749 41BF DE6A'}
keysign.util (WARNING): Cleaned fingerprint to 1354596CD8C790ADD9B27D7090C7974941BFDE6A
keysign.util (DEBUG): Starting HTTP request
keysign.discover (WARNING): Error downloading key, maybe it has been altered in transit
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/requests/models.py", line 379, in prepare_url
    scheme, auth, host, port, path, query, fragment = parse_url(url)
  File "/usr/lib/python3.7/site-packages/urllib3/util/url.py", line 234, in parse_url
    raise LocationParseError(url)
urllib3.exceptions.LocationParseError: Failed to parse: http://[[192.168.10.200]]:9001/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/gnome-keysign/keysign/discover.py", line 49, in start
    key_data = yield threads.deferToThread(self.discovery.find_key, self.userdata)
  File "/usr/lib/python3.7/site-packages/twisted/python/threadpool.py", line 250, in inContext
    result = inContext.theWork()
  File "/usr/lib/python3.7/site-packages/twisted/python/threadpool.py", line 266, in <lambda>
    inContext.theWork = lambda: context.call(ctx, func, *args, **kw)
  File "/usr/lib/python3.7/site-packages/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/lib/python3.7/site-packages/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
  File "/tmp/gnome-keysign/keysign/avahidiscovery.py", line 131, in find_key
    key = super(AvahiKeysignDiscoveryWithMac, self).find_key(userdata)
  File "/tmp/gnome-keysign/keysign/avahidiscovery.py", line 120, in find_key
    downloaded_key = download_key_http(address, port)
  File "/tmp/gnome-keysign/keysign/util.py", line 370, in download_key_http
    data = requests.get(url.geturl(), timeout=5).content
  File "/usr/lib/python3.7/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/lib/python3.7/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 519, in request
    prep = self.prepare_request(req)
  File "/usr/lib/python3.7/site-packages/requests/sessions.py", line 462, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/usr/lib/python3.7/site-packages/requests/models.py", line 313, in prepare
    self.prepare_url(url, params)
  File "/usr/lib/python3.7/site-packages/requests/models.py", line 381, in prepare_url
    raise InvalidURL(*e.args)
requests.exceptions.InvalidURL: Failed to parse: http://[[192.168.10.200]]:9001/
keysign.discover (DEBUG): Returning key: None, success: False, message: 'Error downloading key, maybe it has been altered in transit'
```

**Comment by muelli** [comment]
hm. A few log lines before would be interesting, i.e. https://github.com/gnome-keysign/gnome-keysign/blob/596935d479de05ba050f6da85deb071168f832cb/keysign/avahidiscovery.py#L78

Maybe we're getting this bracketed URL from Avahi already.. hm.

FWIW: It just worked for me.
```
keysign.util (DEBUG): Starting HTTP request
urllib3.connectionpool (DEBUG): Starting new HTTP connection (1): 172.18.157.73
urllib3.connectionpool (DEBUG): http://172.18.157.73:9001 "GET / HTTP/1.1" 200 3854
keysign.util (DEBUG): finished downloading 3854 bytes
```

**Comment by RyuzakiKK** [comment]
The avahi's url seems to be fine:

```
keysign.network.AvahiBrowser (INFO): Found service 'HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949' type '_gnome-keysign._tcp' domain 'local' 
keysign.network.AvahiBrowser (INFO): Found service 'HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949' type '_gnome-keysign._tcp' domain 'local' 
keysign.network.AvahiBrowser (INFO): Service resolved; name: 'HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949', address: 'fe80::2cf:4560:858a:61b2',port: '9001', and txt: '{'version': '1.0.1', 'fingerprint': 'BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949'}'
keysign.avahidiscovery (INFO): discovered something: HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949 fe80::2cf:4560:858a:61b2:9001:BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949
keysign.network.AvahiBrowser (INFO): emitted 'None'
keysign.network.AvahiBrowser (INFO): Service resolved; name: 'HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949', address: '192.168.10.200',port: '9001', and txt: '{'version': '1.0.1', 'fingerprint': 'BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949'}'
keysign.avahidiscovery (INFO): discovered something: HTTP Keyserver BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949 192.168.10.200:9001:BC8A3DA5FA8F68C75EB0A35216DCB5677A3C7949
keysign.network.AvahiBrowser (INFO): emitted 'None'
```

**Comment by muelli** [comment]
hm.
what versions is your requests library?

**Comment by RyuzakiKK** [comment]
python-requests 2.22.0

**Comment by muelli** [comment]
ftr: this is what the flatpaked version uses:
```
$ pip3 freeze
asn1crypto==0.24.0
attrs==18.2.0
autobahn==18.12.1
Automat==0.7.0
Babel==2.6.0
certifi==2018.11.29
cffi==1.11.5
chardet==3.0.4
Click==7.0
constantly==15.1.0
cryptography==2.3.1
dbus-python==1.2.8
docutils==0.14
future==0.17.1
gnome-keysign==1.1.1
gpg==1.12.0
hkdf==0.0.3
humanize==0.5.1
hyperlink==18.0.0
idna==2.8
incremental==17.5.0
lxml==4.3.0
m2r==0.1.12
magic-wormhole==0.11.2
mistune==0.7.4
pyasn1==0.4.5
pyasn1-modules==0.2.3
PyBluez==0.22
pycairo==1.18.0
pycparser==2.19
PyGObject==3.34.0
PyHamcrest==1.9.0
PyNaCl==1.2.1
pyOpenSSL==18.0.0
pytz==2018.9
qrcode==6.0
requests==2.21.0
service-identity==17.0.0
setuptools-scm==1.15.6
six==1.12.0
spake2==0.8
tqdm==4.29.0
Twisted==18.9.0
txaio==18.8.1
txtorcon==18.3.0
urllib3==1.24.1
zope.interface==4.6.0
$ python3 --version
Python 3.7.4
$ 
```

**Comment by RyuzakiKK** [comment]
Apparently this issue appears only with python-requests 2.22.0

I was able to reproduce it in flatpak updating `requests` from 2.21.0 to 2.22.0

**Comment by muelli** [comment]
With urllib1.22 it works.
With urllib > 1.22 it seems to not work.

**Comment by muelli** [comment]
hrm. can you log in this place, what URL gets fed to requests?
```
  File "/tmp/gnome-keysign/keysign/util.py", line 370, in download_key_http
    data = requests.get(url.geturl(), timeout=5).content
```

**Comment by RyuzakiKK** [comment]
> hrm. can you log in this place, what URL gets fed to requests?
> 
> ```
>   File "/tmp/gnome-keysign/keysign/util.py", line 370, in download_key_http
>     data = requests.get(url.geturl(), timeout=5).content
> ```

The `url` variable holds these values:
`ParseResult(scheme='http', netloc='[[192.168.10.200]]:9001', path='/', params='', query='', fragment='')`

And `url.geturl()` is `http://[[192.168.10.200]]:9001/`

**Comment by RyuzakiKK** [comment]
Fixed with #98

**Comment by muelli** [comment]
this should be fixed with https://github.com/gnome-keysign/gnome-keysign/pull/98
